src/utils.ts

```diff
@@ -0,0 +1,30 @@
+import { allPossibleChoices } from './constants';
+
+/**
+ * isBoolean
+ *
+ * Returns true if value is a boolean. This is helpful to determine whether
+ * value is an actual boolean value or just not defined. More exact than a
+ * basic truthy check.
+ *
+ * @param {boolean | null | undefined} val
+ * @returns {boolean}
+ */
+export const isBoolean = (val: boolean | null | undefined) => {
+  return 'boolean' === typeof val;
+};
+
+/**
+ * getRandomHouseResult
+ *
+ * Will return rock, paper or scissors randomly
+ *
+ */
+export const getRandomHouseResult = (uChoice: RockPaperScissors): RockPaperScissors => {
+  // filter all possible choices to exclude what the user has already chosen
```

Comment:
Why exclude what the user has chosen?

Reply:
The reason for doing this is because it takes the draw state out of the equation. I did this here because the designs did not include a screen for the case when there is a draw and in real life we normally just replay straight away so having a draw state seemed kind of redundant to me.